augur/frequencies.py

```diff
@@ -41,72 +64,121 @@ def get_pivots(tps, dt,min_date, max_date):
     return np.arange(first_pivot, last_pivot, dt)
 
 def run(args):
-
-    if args.metadata is None:
-        print("ERROR: meta data with dates is required for frequency estimation")
-        return 1
-
     metadata, columns = read_metadata(args.metadata)
     dates = get_numerical_dates(metadata, fmt='%Y-%m-%d')
     dt = 1.0/args.pivots_per_year
     stiffness = 5.0
 
     if args.tree:
-        from .frequency_estimators import tree_frequencies
         tree = Phylo.read(args.tree, 'newick')
         tps = []
-        for x in tree.get_terminals():
-            x.num_date = np.mean(dates[x.name])
-            tps.append(x.num_date)
-            x.region = metadata[x.name]["region"]
-
-        pivots = get_pivots(tps, dt, args.min_date, args.max_date)
-        frequency_dict = {"pivots":format_frequencies(pivots)}
-
-        # estimate tree frequencies
-        # Omit strains sampled prior to the first pivot from frequency calculations.
-        # (these tend to be reference strains included for phylogenetic context)
-        for region in args.regions:
-            if region=='global':
-                node_filter_func = lambda node: node.num_date >= pivots[0]
-            else:
-                node_filter_func = lambda node: (node.region == region
-                                                and node.num_date >= pivots[0])
-
-            tree_freqs = tree_frequencies(tree, pivots, method='SLSQP',
-                                          node_filter = node_filter_func,
-                                          ws = max(2, tree.count_terminals()//10),
-                                          stiffness = stiffness)
+        for tip in tree.get_terminals():
+            tip.attr = {"num_date": np.mean(dates[tip.name])}
+            tps.append(tip.attr["num_date"])
+
+            # Annotate tips with metadata to enable filtering and weighting of
+            # frequencies by metadata attributes.
+            for key, value in metadata[tip.name].items():
+                tip.attr[key] = value
+
+        if args.method == "diffusion":
+            # estimate tree frequencies
+            pivots = get_pivots(tps, dt, args.min_date, args.max_date)
```

Comment:
it shouldn't take much to harmonize pivot selection across methods.

Reply:
Mostly fixed by the updated `get_pivots` function in `frequency_estimators.py`.